Core/src/TrackFinding/MeasurementSelector.cpp

```diff
@@ -10,19 +10,58 @@
 
 #include "Acts/Definitions/Algebra.hpp"
 #include "Acts/EventData/MeasurementHelpers.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
 
 #include <algorithm>
+#include <cstddef>
+#include <limits>
 
 namespace Acts {
 
 MeasurementSelector::MeasurementSelector()
-    : m_config{{GeometryIdentifier(), MeasurementSelectorCuts{}}} {}
+    : MeasurementSelector({{GeometryIdentifier(), MeasurementSelectorCuts{}}}) {
+}
 
 MeasurementSelector::MeasurementSelector(const MeasurementSelectorCuts& cuts)
-    : m_config{{GeometryIdentifier(), cuts}} {}
+    : MeasurementSelector({{GeometryIdentifier(), cuts}}) {}
+
+MeasurementSelector::MeasurementSelector(const Config& config) {
+  std::vector<InternalConfig::InputElement> tmp;
+  tmp.reserve(config.size());
+  for (std::size_t i = 0; i < config.size(); ++i) {
+    GeometryIdentifier geoID = config.idAt(i);
+    MeasurementSelectorCuts cuts = config.valueAt(i);
+
+    InternalCutBins internalCuts = convertCutBins(cuts);
+    tmp.emplace_back(geoID, std::move(internalCuts));
+  }
+  m_config = InternalConfig(std::move(tmp));
+}
+
+MeasurementSelector::InternalCutBins MeasurementSelector::convertCutBins(
+    const MeasurementSelectorCuts& config) {
+  InternalCutBins cutBins;
+
+  auto getBinOrBackOrMax = [](const auto& vec, std::size_t bin) {
+    using Value = std::remove_reference_t<decltype(vec[0])>;
+    static constexpr Value max = std::numeric_limits<Value>::max();
+    return vec.empty() ? max : (bin < vec.size() ? vec[bin] : vec.back());
+  };
+
+  for (std::size_t bin = 0; bin < config.etaBins.size() + 1; ++bin) {
+    InternalCutBin cuts;
+    cuts.maxAbsTheta = bin < config.etaBins.size()
+                           ? std::acos(std::tanh(config.etaBins[bin]))
+                           : std::numeric_limits<double>::max();
```

Comment:
```suggestion
                           : std::numeric_limits<double>::infinity();
```
we only had to use `max()` to get the generic lambda to work.